framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2744,6 +2756,7 @@ public ParameterizedExecutableType constructorFromUse(NewClassTree tree) {
             // no viewpoint adaptation needed for super invocation
             superCon =
                     AnnotatedTypes.asMemberOf(types, this, type, superCon.getElement(), superCon);
+            con.computeVarargType(superCon);
```

Comment:
Can you add a comment here that explains why the varargs are re-computed here?
Also, maybe the first call, `con.computeVarargType();` should be moved into the `else` branch below?

Reply:
It's also odd that the `else` branch below re-assigns the `con` variable. Is the vararg type adapted by `asMemberOf`?